src/pyenphase/envoy.py

```diff
@@ -298,16 +298,56 @@ def ct_meter_count(self) -> int:
 
     @property
     def consumption_meter_type(self) -> CtType | None:
-        """Return the type of consumption ct meter installed (total or net consumption)."""
+        """Return the type of consumption ct meter installed (total or net consumption or None)."""
         assert self._common_properties is not None, "Call setup() first"  # nosec
         return self._common_properties.consumption_meter_type
 
+    @property
+    def production_meter_type(self) -> CtType | None:
+        """Return the type of production ct meter installed (Production or None)."""
+        assert self._common_properties is not None, "Call setup() first"  # nosec
+        # if 2 ct are installed or only 1 and no consumption then there's a production ct
+        if self.ct_meter_count > 1 or (
+            self.ct_meter_count == 1
+            and self._common_properties.consumption_meter_type is None
+        ):
+            return CtType.PRODUCTION
+        return None
+
     @property
     def phase_mode(self) -> EnvoyPhaseMode | None:
         """Return the phase mode configured for the CT meters (single, split or three)."""
         assert self._common_properties is not None, "Call setup() first"  # nosec
         return self._common_properties.phase_mode
 
+    @property
+    def envoy_model(self) -> str:
```

Comment:
This never changes so we should only generate it once

Reply:
I think you could decorate it with @cached_property 

An older python cached property is usually not used because it has undocumented locking behavior, but since it's rarely going to be accessed, that's probably not a concern for us also the next time assistant release will use python 3.12 so it's even less of a concern since the behavior is gone in that

Reply:
Ok. 

SHould we add Enpower and Encharge info as well based on SupportedFeatures.ENCHARGE and SupportedFeatures.ENPOWER? I'm not into the wording to use for that.

Reply:
> SHould we add Enpower and Encharge info as well based on SupportedFeatures.ENCHARGE and SupportedFeatures.ENPOWER? I'm not into the wording to use for that.

Lets leave that for a future PR if we want to open that door